forc/src/cli/commands/new.rs

```diff
@@ -23,9 +23,15 @@ pub struct Command {
     /// Adding this flag creates an empty workspace.
     #[clap(long)]
     pub workspace: bool,
-    /// Set the package name. Defaults to the directory name
+    /// Set the package name. Defaults to the directory name.
     #[clap(long)]
     pub name: Option<String>,
+    /// The path at which the standard library is located.
+    /// Does not override the std dependency if missing.
+    /// If the path is relative, forc makes it relative w.r.t. the project directory.
+    /// If the path is absolute, forc uses it as is.
+    #[clap(long)]
+    pub stdlib: Option<String>,
```

Comment:
Perhaps we should consider naming this flag `--std-path`, as it only allows overriding the implicit `std` with a path dependency (e.g. you can't use this to specify a `git` URL).

Reply:
This is a very good point! I added a commit to allow URLs as `--stdlib` values. It does not support branches and things like that, but it could be a step in the right direction. If the need arises, we can later modify it to support branches/tags with a syntax like this:
```
forc new foo --stdlib https://github.com/FuelLabs/sway-libs#branch-name
```

WDYT? (If it does not feel right, I can revert that commit and rename `--stdlib` into `--std-path`)

Reply:
I'm not quite sure of the motivation behind #3627, but it seems like the idea is to make it easier for us to create new projects that point to our local version of `std` from our E2E tests directory, as it can be awkward to manually write the correct depth of parent path nesting (i.e. `../../../../../../../../../sway-lib-std`) in the `Forc.toml` due to how deeply those tests are nested - does that sound right?

If so, it seems to me like this flag has a very niche use that is only really relevant to us as Sway compiler devs. I imagine it will be very rare that:

1. Downstream (non-Sway compiler devs) will ever need to override the std dependency manually and
2. For those rare few that do need to specify std, they can just edit the manifest directly like they do for all other dependencies, or use a future `cargo-edit` style command as described in [this issue](https://github.com/FuelLabs/sway/issues/2369) if they need a single shell command, e.g. something like `forc new my-proj && forc add std --path ./path/to/std`.

For these reasons, I'm a bit weary about including the flag in forc's public interface at all, as it may just add noise to the `forc new --help` output for the vast majority of devs.

If we do still want to add it to make our own lives as sway compiler devs easier, maybe we're better off keeping it as simple as possible and stick to supporting just paths? Let me know if I'm missing the point!

Reply:
That was precisely my motivation, you got it right.

Indeed, it makes more sense to go with something like

```
forc new my-proj && forc add std --path ./path/to/std
```

didn't know it was in the works :).

Btw, would it be a good idea to add this *path relativization* feature from this PR (which I'm going to close) to the future `forc add` command?